src/dicom-file.cpp

```diff
@@ -59,20 +59,23 @@ void DicomFile::parse_file() {
     is_valid = i == size;
 }
 
-std::tuple<std::unique_ptr<char[]>,size_t> DicomFile::ApplyFilter(TagFilter filter) {
+std::tuple<nlm::json,std::unique_ptr<char[]>,size_t> DicomFile::ApplyFilter(TagFilter filter) {
+    // todo: implement DOB truncation, and any other special edge cases
     size_t new_size = 0;
     std::unique_ptr<char[]> buffer = nullptr;
+    nlm::json discarded;
     if(is_valid) {
         std::vector<Range> discard_list;
         for (auto &tag: filter) {
             auto iter = elements.find(tag);
             if (iter != elements.end()) {
                 const auto &element = iter->second;
                 discard_list.push_back(element);
+                discarded[tag] = std::string(std::string_view((const char*)data+element.first, element.second-element.first));
             }
         }
         if (discard_list.empty()) {
-            return std::make_tuple(std::move(buffer),new_size);
+            return std::make_tuple(discarded,std::move(buffer),new_size);
```

Comment:
Could probably use this json in the tuple to more conveniently handle PHI in `feat-phi`